woocommerce/woocommerce
# Add customize store transitional screen

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #39721.

This PR implements the customize store transitional screen.

I use the [WP mshot service](https://github.com/Automattic/mShots) to display a preview image. If a site cannot be accessed by mshot server, an error image may appear. It shouldn't happen for WooExpress sites (MVP). In addition to this, It may take a few seconds to load the image for the first time. I think we'll need to improve that or consider embedding the editor iframe instead in a follow-up.

![Screenshot 2023-09-12 at 01 00 40](https://github.com/woocommerce/woocommerce/assets/4344253/34bdb4bf-4e58-41f7-b348-ee55f6532c93)




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use WP 6.3 
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper and enable `customize-store` feature flag
4. Enable xstate inspect via `window.localStorage.setItem( 'xstate_inspect', 'true')`
5. Go to `WooCommerce > Home`
6. Click on `Customize your store` task 
7. Click on `Assembler Hub` button
8. Click on `Done` button
9. A transitional page is shown
10. Confirm the transitional screen matches the design Mrk6SERPZ4KrFHSjM0a8TK-fi-2748%3A260721
11. Click on "Preview store" button -> the store front end should open in a new tab
12. Click on "Go to the Editor" -> should be redirected to the Site Editor
13. Click on "Back to Home"  button
14. Confirm the state is changed to `backToHomescreen` in xstate inspect tab.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add customize store transitional screen

#### Comment 

</details>
